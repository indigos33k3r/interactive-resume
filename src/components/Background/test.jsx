import React from 'react'
import {mount} from 'enzyme'

jest.useFakeTimers()
jest.mock('./utils/backgroundScroll')

describe('Background', () => {
  context('when device is not mobile', () => {
    beforeEach(() => {
      jest.doMock('utils/isMobile', () => () => false)
    })

    it('renders <Background />', () => {
      const component = render()

      expect(component).toMatchSnapshot()
    })

    context('when `props.playerCar` has changed', () => {
      context('when window was scrolled', () => {
        let component, backgroundScroll

        beforeEach(() => {
          global.window.addEventListener = jest.fn((event, fn) => event === 'scroll' && fn())
          backgroundScroll = require('./utils/backgroundScroll').default

          component = render()
          component.setProps({playerCar: 'batmobil'})
          jest.runAllTimers()
          component.update()
        })

        it('removes "background__scroll-info"', () => {
          expect(component).toMatchSnapshot()
        })

        it('calls `backgroundScroll`', () => {
          expect(backgroundScroll).toHaveBeenCalledWith(component.instance().backgroundElmRef.current)
        })
      })
    })
  })

  context('when device is mobile', () => {
    beforeEach(() => {
      jest.doMock('utils/isMobile', () => () => true)
    })

    it('renders <Background /> with mobile controls', () => {
      const component = render()

      expect(component).toMatchSnapshot()
    })

    context('when <button.background__arrow--right /> was clicked', () => {
      let component

      beforeEach(() => {
        component = render()
        component.find('.background__arrow--right').prop('onClick')()
        jest.runAllTimers()
        component.update()
      })

      it('moves <Background /> forward and removes "background__scroll-info"', () => {
        expect(component).toMatchSnapshot()
      })

      context('when <button.background__arrow--left /> was clicked', () => {
        it('moves <Background /> backward', () => {
          component.find('.background__arrow--left').prop('onClick')()
          component.update()

          expect(component).toMatchSnapshot()
        })
      })
    })
  })

  function render(props = {}) {
    const defaultProps = {
      children: 'Some child text'
    }
    const Background = require('.').default

    return mount(<Background {...defaultProps} {...props} />)
  }
})
