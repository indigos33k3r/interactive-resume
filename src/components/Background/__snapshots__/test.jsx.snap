// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Background when device is mobile renders <Background /> with mobile controls 1`] = `
<Background>
  <div
    className="background"
  >
    <div
      className="background__wrapper"
    >
      Some child text
    </div>
    <button
      className="background__arrow background__arrow--left"
      onClick={[Function]}
    />
    <div
      className="background__scroll-info background__scroll-info--mobile"
    >
      Tap these arrows to move
    </div>
    <button
      className="background__arrow background__arrow--right"
      onClick={[Function]}
    />
  </div>
</Background>
`;

exports[`Background when device is mobile when <button.background__arrow--right /> was clicked moves <Background /> forward and removes "background__scroll-info" 1`] = `
<Background>
  <div
    className="background"
  >
    <div
      className="background__wrapper background__wrapper--welcome"
    >
      Some child text
    </div>
    <button
      className="background__arrow background__arrow--left"
      onClick={[Function]}
    />
    <button
      className="background__arrow background__arrow--right"
      onClick={[Function]}
    />
  </div>
</Background>
`;

exports[`Background when device is mobile when <button.background__arrow--right /> was clicked when <button.background__arrow--left /> was clicked moves <Background /> backward 1`] = `
<Background>
  <div
    className="background"
  >
    <div
      className="background__wrapper background__wrapper--tunnel"
    >
      Some child text
    </div>
    <button
      className="background__arrow background__arrow--left"
      onClick={[Function]}
    />
    <button
      className="background__arrow background__arrow--right"
      onClick={[Function]}
    />
  </div>
</Background>
`;

exports[`Background when device is not mobile renders <Background /> 1`] = `
<Background>
  <div
    className="background"
  >
    <div
      className="background__wrapper"
    >
      Some child text
    </div>
    <div
      className="background__scroll-info"
    >
      Scroll to move the car
    </div>
  </div>
</Background>
`;

exports[`Background when device is not mobile when \`props.playerCar\` has changed when window was scrolled removes "background__scroll-info" 1`] = `
<Background
  playerCar="batmobil"
>
  <div
    className="background"
  >
    <div
      className="background__wrapper"
    >
      Some child text
    </div>
  </div>
</Background>
`;
